module/x/peggy/types/key.go

```diff
@@ -43,3 +45,26 @@ func GetValsetConfirmKey(nonce int64, validator sdk.AccAddress) []byte {
 
 	return append(ValsetConfirmKey, append(nonceBytes, []byte(validator)...)...)
 }
+
+// GetClaimKey creates a key with claim type and address first as they have a fix length.
+// We can use this later for prefix scans to find all claims by type,
```

Comment:
in the case that we're doing a prefix scan how do we get the addresses of potentially historical validators when we're looking at past claims?

maybe a handler get the validator set at the claim height?

Reply:
The addresses are encoded into the keys of all claims so that they can be [extracted](https://github.com/alpe/peggy/pull/1/files#diff-bfb43880c36ce0cc8ff43024ac418eceR61). A prefix scan would contain a range that covers the path. I don't think that the fix lenght nonce is a good idea anymore. We should make this a string to match what the ETH side emits. 
If you know the common queries already then I may prepare for second indexes instead.